### ISSUE TITLE
ci: Windows build fails intermittently when downloading (Failed to download from mirror set)

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#23107]
It seems that sometimes a download will fail, but succeeds on the retry, however the full build will still be marked as failed:

https://cirrus-ci.com/task/4800755426131968?logs=build#L31

```
  A suitable version of powershell-core was not found (required v7.1.0). Downloading portable powershell-core v7.1.0...
  Downloading powershell-core...
    https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip -> C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\downloads\PowerShell-7.1.0-win-x86.zip
  Failed to download from mirror set:
  https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip: WinHttpReceiveResponse() failed: 12030
  
C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\vcpkg.targets(94,5): error MSB3073: The command ""C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\\" "--x-manifest-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\\" "--x-install-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\\" " exited with code 1. [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\libbitcoinconsensus\libbitcoinconsensus.vcxproj]
  Waiting to take filesystem lock on C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\.vcpkg-root...
  C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc
  Installing vcpkg dependencies to C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\
  "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\scripts\buildsystems\msbuild\..\..\..\\" "--x-manifest-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\\" "--x-install-root=C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\vcpkg_installed\\" 
  Detecting compiler hash for triplet x64-windows-static...
  A suitable version of powershell-core was not found (required v7.1.0). Downloading portable powershell-core v7.1.0...
  Downloading powershell-core...
    https://github.com/PowerShell/PowerShell/releases/download/v7.1.0/PowerShell-7.1.0-win-x86.zip -> C:\Users\ContainerAdministrator\AppData\Local\Temp\vcpkg\downloads\PowerShell-7.1.0-win-x86.zip
  Extracting powershell-core...

**Comment by hebasto** [comment]
Is it a `vcpkg` issue?

**Comment by MarcoFalke** [comment]
Maybe just run the command a second time if the first one fails?

**Comment by hebasto** [comment]
Addressed in #23215.
